File: frontend/src/lib/types/oast.ts
export interface OastRead {
	mode: string;
	server: string | null;
	wait_seconds: number;
	public_acknowledged: boolean;
	token_set: boolean;
	public_servers: string[];
	checks: number;
	reason: string | null;
	last_interaction_at: string | null;
	interactions: number;
}

export interface OastUpdate {
	mode?: string | null;
	server?: string | null;
	wait_seconds?: number | null;
	public_acknowledged?: boolean | null;
}

export interface OastTest {
	ok: boolean;
	detail: string;
	address: string | null;
	status_code: number | null;
}

File: shared/models/oast.py
"""Out-of-band settings as the API reads and writes them."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict
from pydantic import Field as PydanticField

from shared.definitions.oast import (
    MAX_SERVER_LENGTH,
    MAX_WAIT_SECONDS,
    MIN_WAIT_SECONDS,
    OastMode,
)


class OastRead(BaseModel):
    mode: str = OastMode.OFF.value
    server: str | None = None
    wait_seconds: int = 0
    public_acknowledged: bool = False
    token_set: bool = False
    public_servers: list[str] = PydanticField(default_factory=list)
    checks: int = 0
    reason: str | None = None
    last_interaction_at: datetime | None = None
    interactions: int = 0


class OastUpdate(BaseModel):
    model_config = ConfigDict(extra="forbid")

    mode: OastMode | None = None
    server: str | None = PydanticField(default=None, max_length=MAX_SERVER_LENGTH)
    wait_seconds: int | None = PydanticField(
        default=None, ge=MIN_WAIT_SECONDS, le=MAX_WAIT_SECONDS
    )
    public_acknowledged: bool | None = None


class OastTest(BaseModel):
    ok: bool = False
    detail: str = ""
    address: str | None = None
    status_code: int | None = None


__all__ = ["OastRead", "OastTest", "OastUpdate"]
